ui/lib/CancelStream.tsx

```diff
@@ -62,42 +57,12 @@ const CancelStream: React.FC<CancelStreamProps> = (props) => {
     }
   };
 
-  // Fetch starttime
-  useEffect(() => {
-    const getStartTime = async () => {
-      try {
-        const streamInfo = await props.stream.streamInfo();
-        setStartTime(Number(streamInfo.start_time));
-        if (streamInfo && streamInfo.start_time !== 0n) {
-          clearInterval(interval);
-        }
-      } catch (error) {
-        console.error("Error getting stream token");
-      }
-    };
-    const interval = setInterval(getStartTime, 1000);
-    return () => clearInterval(interval);
-  }, [startTime]);
-
-  // Calculate the time in seconds before a stream can be cancelled
-  const timeBeforeCancellability = Math.max(
-    0,
-    startTime + minStreamLife - currentTime,
-  );
-
   return (
     <div className="stream-container">
-      {minStreamLife === null ? (
-        <div className="cancel-stream-label-loading">
-          Fetching minimum life...
-        </div>
-      ) : !isButtonEnabled &&
-        !inProgress &&
-        startTime !== 0 &&
-        Number(timeBeforeCancellability) !== 0 ? (
+      {!isButtonEnabled && !inProgress && timeBeforeCancellability > 0 ? (
         <>
           <div className="cancel-stream-label-min-life">
-            Deployment cannot be cancelled yet: its minimum life is
+            Simulation cannot be cancelled yet: its minimum life is
```

Comment:
I think this label should be configurable (maybe default is this)
```suggestion
            Stream cannot be cancelled yet: its minimum life is
```